xarray/namedarray/core.py

```diff
@@ -720,6 +731,153 @@ def sizes(self) -> dict[_Dim, _IntOrUnknown]:
         """Ordered mapping from dimension names to lengths."""
         return dict(zip(self.dims, self.shape))
 
+    def chunk(
+        self,
+        chunks: (
+            int
+            | Literal["auto"]
+            | tuple[int, ...]
+            | tuple[tuple[int, ...], ...]
+            | Mapping[Any, None | int | tuple[int, ...]]
+        ) = {},
+        name: str | None = None,
+        lock: bool | None = None,
+        inline_array: bool | None = None,
```

Comment:
Talking about other API that isn't satisfying - `name`, `lock`, `inline_array` are all dask-specific, so should ideally be folded into `from_array_kwargs`